rosserial_arduino/src/ros_lib/ArduinoHardware.h

```diff
@@ -58,16 +58,17 @@
   // Stm32duino Maple mini USB Serial Port
   #define SERIAL_CLASS USBSerial
 #else 
-  #include <HardwareSerial.h>  // Arduino AVR
-  #if defined(USE_USBCDC)
-    // ESP32-S2 ESP32-C3
-    #define SERIAL_CLASS USBCDC
-  #elif defined(USE_HWCDC)
-    // ESP32-S3
-    #define SERIAL_CLASS HWCDC
-  #else
+
+  #if ARDUINO_USB_CDC_ON_BOOT 
+    #if ARDUINO_USB_MODE 
+      #define SERIAL_CLASS HWCDC
+    #else
+      #define SERIAL_CLASS USBCDC
+    #endif  
+  #else  
+    #include <HardwareSerial.h>  // Arduino AVR
```

Comment:
I think you can include `HardwareSerial.h` unconditionally? `Serial` is an instance of the correct type then, as defined in
https://github.com/espressif/arduino-esp32/blob/cf448906b3836fbe9368934713b697469254c62f/cores/esp32/HardwareSerial.h#L346-L366

And that is also what gets assigned to `SERIAL_CLASS *iostream` in
https://github.com/Intermodalics/rosserial/blob/112cebcbd13076c90bb5d2506044a64d7f412724/rosserial_arduino/src/ros_lib/ArduinoHardware.h#L88
and hence has to work either way unless you would add the preprocessor conditions there, too.

So
```cpp
#define SERIAL_CLASS decltype(Serial)
```
or
```cpp
typedef decltype(Serial) SERIAL_CLASS;
```
should work...

Reply:
Tested it and it is still the same, it froze. The good news is that we can use `HWCDC` and also flash it using esptool libraries

Reply:
> I can test it but this is the way that we have to initialize the baud rate, right?

There is no baud rate in [CDC mode](https://en.wikipedia.org/wiki/USB_communications_device_class), either one. The argument in some calls is only for compatibility with `Serial.begin(baudrate)` calls in Arduino code if `Serial` would refer to a CDC device. Same on the Linux side.